### DISCUSSION TITLE
Failover (when master fails)

### DISCUSSION THREAD

**Discussion by MikeJakubik** [postgresml/pgcat#468] [General]
How does pgcat handle failover if the master server dies? Cant seem to find any info on this.

**Comment by levkk** [comment]
It doesn't. Currently, you need to use another primary failover tool, e.g. Patroni and update pgcat config accordingly.

**Reply by MTRNord** [reply]
Are there any plans on supporting this eventually and/or already someone that might show an example setup on how to do this in a sensible way?

**Reply by levkk** [reply]
Not at the moment. We use external software to detect the failure of the primary and update the config in PgCat accordingly. I think it's a good idea to add some kind of primary healthcheck and also add a standby into the config. If the primary fails, we should make the standby the primary and route write traffic to it.

**Reply by PatrickBzh** [reply]
Hello, 
Is there some news on this subject since sept. 2023 ?
